### PR TITLE
[performance](executor) optimize time_round function

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

1. avoid convert_to_full_column_if_const 
2. do not check valid
3. optimize the part where the period is less than 12 by compiler.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.48 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162174467 Bytes

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

File: be/src/vec/functions/function_datetime_floor_ceil.cpp [hunk 1]
```diff
@@ -150,51 +150,67 @@ class FunctionDateTimeFloorCeil : public IFunction {
...
+                if (arg1_const && arg2_const) {
+                    Field arg1, arg2;
+                    arg1_col->get(0, arg1);
+                    arg2_col->get(0, arg2);
+                    // time_round(datetime,const(period) , const(origin))
+                    Impl::template vector_const_const<NativeType, DeltaValueType>(
+                            sources->get_data(), arg1.get<Int32>(), arg2.get<NativeType>(),
+                            col_to->get_data(), null_map->get_data());
+
+                } else {
```
Comment (HappenLee) [inline be/src/vec/functions/function_datetime_floor_ceil.cpp:202]: must dispose the 1 const 1 not ？

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.16 seconds
 stream load tsv:          546 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162038011 Bytes

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run external

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
This optimization is too much of a trick and not conducive to future maintenance. We are discussing whether to introduce a new simplified function.

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.27 seconds
 stream load tsv:          531 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17162153036 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.48 seconds
 stream load tsv:          532 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17162278349 Bytes

**Comment by wm1581066** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**